src/query/plan/operator.cpp

```diff
@@ -4674,22 +4675,31 @@ auto ToOptionalString(ExpressionEvaluator *evaluator, Expression *expression) ->
   return std::nullopt;
 };
 
-TypedValue CsvRowToTypedList(csv::Reader::Row &row) {
+TypedValue CsvRowToTypedList(csv::Reader::Row &row, std::optional<utils::pmr::string> &nullif) {
   auto *mem = row.get_allocator().GetMemoryResource();
   auto typed_columns = utils::pmr::vector<TypedValue>(mem);
   typed_columns.reserve(row.size());
   for (auto &column : row) {
-    typed_columns.emplace_back(std::move(column));
+    if (!nullif.has_value() || column != nullif.value()) {
+      typed_columns.emplace_back(std::move(column));
+    } else {
+      typed_columns.emplace_back(TypedValue(mem));
```

Comment:
```suggestion
      typed_columns.emplace_back(mem);
```

Since you are using `emplace_back` on vector, you can pass just `mem` and it should construct `TypedValue` by passing `mem` to TypedValue `ctor`. I am not sure if this does a necessary copy, but I think yes.

I checked it does 1 move and 1 copy unnecessary.
https://godbolt.org/z/sbf38xYhd

Reply:
I was getting some weird behaviour with explicit constructor on memory but I'm going to check it once again

Reply:
Fixed with no arguments in `emplace_back()`